federation/schema/normalize.go

```diff
@@ -0,0 +1,47 @@
+package federation
+
+
+import (
+	"fmt"
+
+	"github.com/samsarahq/thunder/graphql"
+)
+
+
+func collectTypes(typ graphql.Type, types map[graphql.Type]string) error {
```

Comment:
// collectTypes finds all types reachable from typ and stores them in types as a map from type to name.